sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/PartitionMetadata.scala

```diff
@@ -110,14 +115,24 @@ private[cosmos] case class PartitionMetadata
 
   def getWeightedLsnGap: Long = {
     val progressFactor = math.max(this.latestLsn - this.startLsn, 0)
-    val averageItemsPerLsn = if (this.documentCount == 0) {
-      1d
+    if (progressFactor == 0) {
```

Comment:
Any reason to special case this? The result at the end would still be 0, wouldn't it?

Reply:
Yes-  when startLsn < latestLsn we ensure that the weighted gap (LSN gap * avg. # docs per LSN) is at least 1 - if there is no gap we return 0